src/irmin-pack/irmin_pack.ml

```diff
@@ -700,43 +700,46 @@ module Pack (K : Irmin.Hash.S) = struct
 
     let buffer_for_hash = Bytes.create K.hash_size
 
+    let io_read_and_decode ~off ~len t =
+      let buf = Bytes.create len in
+      let n = IO.read t.pack.block ~off buf in
+      assert (n = len);
+      let hash off =
+        let n = IO.read t.pack.block ~off buffer_for_hash in
+        assert (n = K.hash_size);
+        let _, v =
+          Irmin.Type.decode_bin ~headers:false K.t
+            (* the copy is important here *)
+            (Bytes.to_string buffer_for_hash)
+            0
+        in
+        v
+      in
+      let dict = Dict.find t.pack.dict in
+      V.decode_bin ~hash ~dict (Bytes.unsafe_to_string buf) 0
+
     let unsafe_find t k =
       Log.debug (fun l -> l "[pack] find %a" pp_hash k);
       stats.pack_finds <- succ stats.pack_finds;
       match Tbl.find t.staging k with
       | v ->
-          Lru.add t.lru k v;
-          Some v
+        Lru.add t.lru k v;
```

Comment:
this whitespace change is a bit strange, are you using ocamlformat?

Reply:
I've been using `ocp-indent`, just because it came stock. Will sort out `ocamlformat` now...